src/cli/snp.rs

```diff
@@ -0,0 +1,260 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::backend::sev::Firmware;
+
+use anyhow::{anyhow, Context, Result};
+use std::fs::{self, remove_file};
+use std::io::{self, ErrorKind, Read, Seek};
+use std::os::unix::fs::OpenOptionsExt;
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+/// SNP-specific functionality
+#[derive(StructOpt, Debug)]
+pub enum Command {
+    /// SNP VCEK related commands
+    Vcek(VcekCommand),
+}
+
+#[derive(StructOpt, Debug)]
+pub enum VcekCommand {
+    /// Print the VCEK certificate for this platform to stdout in PEM format
+    Show,
+    /// Print the VCEK certificate cache file used
+    ShowFile,
+    /// Download the VCEK certificate for this platform and save it to a cache file in `/var/cache/amd-sev/`
+    /// or `$XDG_CACHE_HOME` or `$HOME/.cache/`
+    Update,
+}
+
+pub fn run(cmd: Command) -> Result<()> {
+    match cmd {
+        Command::Vcek(VcekCommand::Show) => {
+            let mut reader = get_vcek_reader()?;
+            io::copy(&mut reader, &mut io::stdout())?;
+            Ok(())
+        }
+        Command::Vcek(VcekCommand::ShowFile) => {
+            let (path, _) = get_vcek_reader_with_paths(paths())?;
```

Comment:
Does this download and write to fs if file is not present?

Reply:
fixed: not anymore.